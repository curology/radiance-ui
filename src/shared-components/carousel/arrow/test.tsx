import React from 'react';
import { render } from 'src/tests/testingLibraryHelpers';

import Arrow from './index';

describe('<Arrow />', () => {
  describe('UI snapshots', () => {
    it('renders with props', () => {
      const { container } = render(
        <Arrow prev disabled={false} onClick={() => undefined} />,
        { withPortalContainer: true },
      );

      expect(container).toMatchSnapshot();
    });

    it('renders with bottom right alignment', () => {
      const { container } = render(
        <Arrow
          bottomRightAlignedArrows
          prev
          disabled
          onClick={() => undefined}
        />,
        { withPortalContainer: true },
      );

      expect(container).toMatchSnapshot();
    });
  });
});
