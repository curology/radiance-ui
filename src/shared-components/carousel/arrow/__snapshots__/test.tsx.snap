// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Arrow /> UI snapshots renders with bottom right alignment 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0) scale(1, 1);
    -moz-transform: translate3d(0, 0, 0) scale(1, 1);
    -ms-transform: translate3d(0, 0, 0) scale(1, 1);
    transform: translate3d(0, 0, 0) scale(1, 1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px, 0, 0) scale(1, 1);
    -moz-transform: translate3d(12px, 0, 0) scale(1, 1);
    -ms-transform: translate3d(12px, 0, 0) scale(1, 1);
    transform: translate3d(12px, 0, 0) scale(1, 1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -moz-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -ms-transform: translate3d(16px, 0, 0) scale(.85, .85);
    transform: translate3d(16px, 0, 0) scale(.85, .85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0);
    -moz-transform: translate3d(16px, 0, 0);
    -ms-transform: translate3d(16px, 0, 0);
    transform: translate3d(16px, 0, 0);
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0) scale(1, 1);
    -moz-transform: translate3d(0, 0, 0) scale(1, 1);
    -ms-transform: translate3d(0, 0, 0) scale(1, 1);
    transform: translate3d(0, 0, 0) scale(1, 1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px, 0, 0) scale(1, 1);
    -moz-transform: translate3d(12px, 0, 0) scale(1, 1);
    -ms-transform: translate3d(12px, 0, 0) scale(1, 1);
    transform: translate3d(12px, 0, 0) scale(1, 1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -moz-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -ms-transform: translate3d(16px, 0, 0) scale(.85, .85);
    transform: translate3d(16px, 0, 0) scale(.85, .85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0);
    -moz-transform: translate3d(16px, 0, 0);
    -ms-transform: translate3d(16px, 0, 0);
    transform: translate3d(16px, 0, 0);
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0) scale(1, 1);
    -moz-transform: translate3d(0, 0, 0) scale(1, 1);
    -ms-transform: translate3d(0, 0, 0) scale(1, 1);
    transform: translate3d(0, 0, 0) scale(1, 1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px, 0, 0) scale(1, 1);
    -moz-transform: translate3d(12px, 0, 0) scale(1, 1);
    -ms-transform: translate3d(12px, 0, 0) scale(1, 1);
    transform: translate3d(12px, 0, 0) scale(1, 1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -moz-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -ms-transform: translate3d(16px, 0, 0) scale(.85, .85);
    transform: translate3d(16px, 0, 0) scale(.85, .85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0);
    -moz-transform: translate3d(16px, 0, 0);
    -ms-transform: translate3d(16px, 0, 0);
    transform: translate3d(16px, 0, 0);
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0) scale(1, 1);
    -moz-transform: translate3d(0, 0, 0) scale(1, 1);
    -ms-transform: translate3d(0, 0, 0) scale(1, 1);
    transform: translate3d(0, 0, 0) scale(1, 1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px, 0, 0) scale(1, 1);
    -moz-transform: translate3d(12px, 0, 0) scale(1, 1);
    -ms-transform: translate3d(12px, 0, 0) scale(1, 1);
    transform: translate3d(12px, 0, 0) scale(1, 1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -moz-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -ms-transform: translate3d(16px, 0, 0) scale(.85, .85);
    transform: translate3d(16px, 0, 0) scale(.85, .85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0);
    -moz-transform: translate3d(16px, 0, 0);
    -ms-transform: translate3d(16px, 0, 0);
    transform: translate3d(16px, 0, 0);
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0) scale(1, 1);
    -moz-transform: translate3d(0, 0, 0) scale(1, 1);
    -ms-transform: translate3d(0, 0, 0) scale(1, 1);
    transform: translate3d(0, 0, 0) scale(1, 1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px, 0, 0) scale(1, 1);
    -moz-transform: translate3d(12px, 0, 0) scale(1, 1);
    -ms-transform: translate3d(12px, 0, 0) scale(1, 1);
    transform: translate3d(12px, 0, 0) scale(1, 1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -moz-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -ms-transform: translate3d(16px, 0, 0) scale(.85, .85);
    transform: translate3d(16px, 0, 0) scale(.85, .85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0);
    -moz-transform: translate3d(16px, 0, 0);
    -ms-transform: translate3d(16px, 0, 0);
    transform: translate3d(16px, 0, 0);
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0) scale(1, 1);
    -moz-transform: translate3d(0, 0, 0) scale(1, 1);
    -ms-transform: translate3d(0, 0, 0) scale(1, 1);
    transform: translate3d(0, 0, 0) scale(1, 1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px, 0, 0) scale(1, 1);
    -moz-transform: translate3d(12px, 0, 0) scale(1, 1);
    -ms-transform: translate3d(12px, 0, 0) scale(1, 1);
    transform: translate3d(12px, 0, 0) scale(1, 1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -moz-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -ms-transform: translate3d(16px, 0, 0) scale(.85, .85);
    transform: translate3d(16px, 0, 0) scale(.85, .85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0);
    -moz-transform: translate3d(16px, 0, 0);
    -ms-transform: translate3d(16px, 0, 0);
    transform: translate3d(16px, 0, 0);
  }
}

.emotion-0 {
  display: block;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: flex-end;
  align-self: flex-end;
  margin: 0.5rem;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  margin-left: auto;
}

.emotion-0 button {
  background-color: none;
  border-color: #332E54;
  color: #332E54;
}

.emotion-0 button:hover {
  color: #332E54;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-4 {
  color: #524D6E;
  font-size: 0.75rem;
  line-height: 1.67;
  font-weight: bold;
  letter-spacing: 1px;
  text-transform: uppercase;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  display: block;
  margin: 0;
  min-height: 52px;
  opacity: 1;
  padding: 0 1.5rem;
  position: relative;
  -webkit-transition: all 350ms ease-in-out;
  transition: all 350ms ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #EDEDF0;
  border-color: #EDEDF0;
  color: #C3C0CD;
  cursor: not-allowed;
  fill: #C3C0CD;
  min-width: 208px;
  max-width: 325px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  border-radius: 50%;
  max-width: unset;
  min-height: unset;
  min-width: unset;
  height: 48px;
  width: 48px;
  padding: 0;
}

.emotion-4:hover {
  -webkit-transition: all 350ms ease-in-out;
  transition: all 350ms ease-in-out;
}

.emotion-4:active,
.emotion-4:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-4>svg {
  opacity: 1;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
  margin: 0 auto;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  position: absolute;
  bottom: 0;
  right: 0;
  top: 0;
  margin-top: -6px;
  width: 38px;
  opacity: 0;
  width: 36px;
  margin: -3px -3px 0 0;
}

.emotion-7 span {
  border-radius: 50%;
  display: inline-block;
  height: 4px;
  width: 4px;
  position: absolute;
}

.emotion-7 span:nth-of-type(1) {
  -webkit-animation: animation-0 1750ms infinite linear;
  animation: animation-0 1750ms infinite linear;
}

.emotion-7 span:nth-of-type(2) {
  -webkit-animation: animation-0 1750ms infinite linear -300ms;
  animation: animation-0 1750ms infinite linear -300ms;
}

.emotion-7 span:nth-of-type(3) {
  -webkit-animation: animation-0 1750ms infinite linear -600ms;
  animation: animation-0 1750ms infinite linear -600ms;
}

.emotion-7 span {
  background-color: #FFFFFF;
}

<div
  id="reactPortalSection"
>
  <div
    class="emotion-0 emotion-1"
    disabled=""
  >
    <div
      class="emotion-2 emotion-3"
    >
      <button
        class="emotion-4 emotion-5"
        disabled=""
        type="button"
      >
        <svg
          class="emotion-6"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 7.21H2.41L8.74.89 7.85 0 0 7.86l8.18 7.87.87-.9-6.63-6.37H16V7.21z"
            fill="currentColor"
          />
        </svg>
        <div
          class="emotion-7 emotion-8"
          disabled=""
        >
          <div>
            <span />
            <span />
            <span />
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`<Arrow /> UI snapshots renders with props 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0) scale(1, 1);
    -moz-transform: translate3d(0, 0, 0) scale(1, 1);
    -ms-transform: translate3d(0, 0, 0) scale(1, 1);
    transform: translate3d(0, 0, 0) scale(1, 1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px, 0, 0) scale(1, 1);
    -moz-transform: translate3d(12px, 0, 0) scale(1, 1);
    -ms-transform: translate3d(12px, 0, 0) scale(1, 1);
    transform: translate3d(12px, 0, 0) scale(1, 1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -moz-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -ms-transform: translate3d(16px, 0, 0) scale(.85, .85);
    transform: translate3d(16px, 0, 0) scale(.85, .85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0);
    -moz-transform: translate3d(16px, 0, 0);
    -ms-transform: translate3d(16px, 0, 0);
    transform: translate3d(16px, 0, 0);
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0) scale(1, 1);
    -moz-transform: translate3d(0, 0, 0) scale(1, 1);
    -ms-transform: translate3d(0, 0, 0) scale(1, 1);
    transform: translate3d(0, 0, 0) scale(1, 1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px, 0, 0) scale(1, 1);
    -moz-transform: translate3d(12px, 0, 0) scale(1, 1);
    -ms-transform: translate3d(12px, 0, 0) scale(1, 1);
    transform: translate3d(12px, 0, 0) scale(1, 1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -moz-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -ms-transform: translate3d(16px, 0, 0) scale(.85, .85);
    transform: translate3d(16px, 0, 0) scale(.85, .85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0);
    -moz-transform: translate3d(16px, 0, 0);
    -ms-transform: translate3d(16px, 0, 0);
    transform: translate3d(16px, 0, 0);
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0) scale(1, 1);
    -moz-transform: translate3d(0, 0, 0) scale(1, 1);
    -ms-transform: translate3d(0, 0, 0) scale(1, 1);
    transform: translate3d(0, 0, 0) scale(1, 1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px, 0, 0) scale(1, 1);
    -moz-transform: translate3d(12px, 0, 0) scale(1, 1);
    -ms-transform: translate3d(12px, 0, 0) scale(1, 1);
    transform: translate3d(12px, 0, 0) scale(1, 1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -moz-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -ms-transform: translate3d(16px, 0, 0) scale(.85, .85);
    transform: translate3d(16px, 0, 0) scale(.85, .85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0);
    -moz-transform: translate3d(16px, 0, 0);
    -ms-transform: translate3d(16px, 0, 0);
    transform: translate3d(16px, 0, 0);
  }
}

.emotion-0 {
  position: absolute;
  top: 50%;
  z-index: 2000;
  -webkit-transform: translate(0%, -50%);
  -moz-transform: translate(0%, -50%);
  -ms-transform: translate(0%, -50%);
  transform: translate(0%, -50%);
  display: block;
  left: 0.5rem;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-4 {
  color: #524D6E;
  font-size: 0.75rem;
  line-height: 1.67;
  font-weight: bold;
  letter-spacing: 1px;
  text-transform: uppercase;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  display: block;
  margin: 0;
  min-height: 52px;
  opacity: 1;
  padding: 0 1.5rem;
  position: relative;
  -webkit-transition: all 350ms ease-in-out;
  transition: all 350ms ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-width: 1px;
  border-color: #EDEDF0;
  background-color: #FFFFFF;
  color: #332E54;
  fill: #332E54;
  box-shadow: 0px 8px 24px rgba(52, 51, 82, 0.05);
  min-width: 208px;
  max-width: 325px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  border-radius: 50%;
  max-width: unset;
  min-height: unset;
  min-width: unset;
  height: 48px;
  width: 48px;
  padding: 0;
}

.emotion-4:hover {
  -webkit-transition: all 350ms ease-in-out;
  transition: all 350ms ease-in-out;
}

.emotion-4:active,
.emotion-4:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-4:hover {
  box-shadow: 0px 8px 24px rgba(52, 51, 82, 0.10);
}

.emotion-4:hover {
  -webkit-transition: all 350ms ease-in-out;
  transition: all 350ms ease-in-out;
  opacity: 1;
}

.emotion-4>svg {
  opacity: 1;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
  margin: 0 auto;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  position: absolute;
  bottom: 0;
  right: 0;
  top: 0;
  margin-top: -6px;
  width: 38px;
  opacity: 0;
  width: 36px;
  margin: -3px -3px 0 0;
}

.emotion-7 span {
  background-color: #332E54;
  border-radius: 50%;
  display: inline-block;
  height: 4px;
  width: 4px;
  position: absolute;
}

.emotion-7 span:nth-of-type(1) {
  -webkit-animation: animation-0 1750ms infinite linear;
  animation: animation-0 1750ms infinite linear;
}

.emotion-7 span:nth-of-type(2) {
  -webkit-animation: animation-0 1750ms infinite linear -300ms;
  animation: animation-0 1750ms infinite linear -300ms;
}

.emotion-7 span:nth-of-type(3) {
  -webkit-animation: animation-0 1750ms infinite linear -600ms;
  animation: animation-0 1750ms infinite linear -600ms;
}

<div
  id="reactPortalSection"
>
  <div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <button
        class="emotion-4 emotion-5"
        type="button"
      >
        <svg
          class="emotion-6"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 7.21H2.41L8.74.89 7.85 0 0 7.86l8.18 7.87.87-.9-6.63-6.37H16V7.21z"
            fill="currentColor"
          />
        </svg>
        <div
          class="emotion-7 emotion-8"
        >
          <div>
            <span />
            <span />
            <span />
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;
